editor/tests/org.fusesource.ide.camel.editor.tests.integration/src/main/java/org/fusesource/ide/camel/editor/integration/globalconfiguration/dataformat/wizards/NewDataFormatWizardIT.java

```diff
@@ -86,7 +110,57 @@ public void testCreation() throws CoreException, IOException {
 		final String id = dataFormat.getName() + "-id";
 		Element dataFormatNode = newDataFormatWizard.createDataFormatNode(dataFormat, id);
 		new CamelGlobalConfigEditor(null).addDataFormat(camelFile, dataFormatNode);
+		
+		// Check that element just created has been correctly initialized
+		check(id, camelFile);
+		// Check that Model is valid after reloading from the filesystem
+		final CamelIOHandler camelIOHandler = new CamelIOHandler();
+		camelIOHandler.setDocument(camelFile.getDocument());
+		camelIOHandler.saveCamelModel(camelFile, camelFile.getResource().getLocation().toFile(), new NullProgressMonitor());
+		CamelFile reloadedCamelFile = camelIOHandler.loadCamelModel(camelFile.getResource(), new NullProgressMonitor());
+		check(id, reloadedCamelFile);
+	}
+
+	@Test
+	public void testCreationWithDeps() throws CoreException, IOException, InterruptedException, InvocationTargetException {
+		final String id = dataFormat.getName() + "-id2";
+		final String projectName = dataFormat.getName() + "-project";
+		final String camelFilePath = "src/main/resources/META-INF/spring/camel-context.xml";
+		
+		NewProjectMetaData metadata;
+		metadata = new NewProjectMetaData();
+		metadata.setProjectName(projectName);
+		metadata.setLocationPath(null);
+		metadata.setCamelVersion(camelVersion);
+		metadata.setTargetRuntime(null);
+		metadata.setDslType(CamelDSLType.SPRING);
+		metadata.setBlankProject(true);
+		metadata.setTemplate(null);
+		
+		new ProgressMonitorDialog(
+				Display.getDefault().getActiveShell()).run(false, true, 
+						new FuseIntegrationProjectCreatorRunnable(metadata));
+
+		new ProgressMonitorDialog(Display.getDefault().getActiveShell()).run(false, true, new FuseIntegrationProjectCreatorRunnable(metadata));
+
+		IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(projectName);
+
+		assertThat(project.exists()).describedAs("The project "+ project.getName()+ " doesn't exist.").isTrue();
+		CamelEditorUIActivator.pluginLog().logInfo("Project created: "+projectName);
+		final IFile camelResource = project.getFile(camelFilePath);
+		assertThat(camelResource.exists()).isTrue();
+
+		// TODO: wait for all build job to finish?
+		waitJob();
+
+		CamelModel camelModel = CamelModelFactory.getModelForVersion(metadata.getCamelVersion());
+
+		final CamelFile camelFile = new CamelFile(camelResource);
```

Comment:
the Camel File need to be initialized. The best is to use new CamelIOHandler().loadCamelModel(camelResource, new NullProgressMonitor()); which will take care of loading everything properly